typescript/src/lib/types/Augments.d.ts

```diff
@@ -124,6 +128,14 @@ declare module '@sapphire/framework' {
 			content: string
 		): boolean;
 		emit(event: Events.GuildAnnouncementError, message: Message, channel: TextChannel, role: Role, content: string, error: any): boolean;
+		emit(
+			event: Events.GuildMessageLog,
+			guild: Guild,
+			channelId: string | Nullish,
+			key: keyof GuildEntity,
```

Comment:
If you can make a utility type that filters everything that starts with `channelsLogs`, that would be awesome. Possibly with a combination of `Exclude<>` and string template tags?

---
---

Alternatively we can use this utility type:

```ts
/**
 * PickByValueExact
 * @desc From `T` pick a set of properties by value matching exact `ValueType`.
 * @example
 *   type Props = { req: number; reqUndef: number | undefined; opt?: string; };
 *
 *   // Expect: { req: number }
 *   type Props = PickByValueExact<Props, number>;
 *   // Expect: { reqUndef: number | undefined; }
 *   type Props = PickByValueExact<Props, number | undefined>;
 */
export type PickByValueExact<T, ValueType> = Pick<
	T,
	{
		[Key in keyof T]-?: [ValueType] extends [T[Key]] ? ([T[Key]] extends [ValueType] ? Key : never) : never;
	}[keyof T]
>;
```

Based on: https://github.com/piotrwitek/utility-types/blob/df2502ef504c4ba8bd9de81a45baef112b7921d0/src/mapped-types.ts#L196-L216

```suggestion
			key: keyof PickByValueExact<GuildEntity, string | undefined | null>,
```

It's not perfect, but it reduces the total amount of possible keys from 212 to 34. It will remove something like `getLanguage` from the union.

Screenshot based on main:
![image](https://user-images.githubusercontent.com/4019718/125696118-32ca42e2-7dda-4a11-abc6-e1224a2dff5f.png)

Reply:
We already have `PickByValue`.